azure-kusto-ingest/azure/kusto/ingest/ingest_client.py

```diff
@@ -53,18 +52,18 @@ def ingest_from_file(self, file_descriptor: Union[FileDescriptor, str], ingestio
         """
         containers = self._get_containers()
 
-        if isinstance(file_descriptor, FileDescriptor):
-            descriptor = file_descriptor
-        else:
-            descriptor = FileDescriptor(file_descriptor)
-
-        should_compress = not descriptor.is_compressed and ingestion_properties.format.compressible
-
-        with descriptor.open(should_compress) as stream:
-            blob_descriptor = self._upload_blob(containers, descriptor, ingestion_properties, stream)
-            result = self.ingest_from_blob(blob_descriptor, ingestion_properties=ingestion_properties)
-
-        return result
+        file_descriptor, should_compress = BaseIngestClient._prepare_file(file_descriptor, ingestion_properties)
+        with file_descriptor.open(should_compress) as stream:
+            blob_descriptor = BlobDescriptor.from_different_descriptor(
+                containers,
+                file_descriptor,
+                ingestion_properties.database,
+                ingestion_properties.table,
+                stream,
+                self._proxy_dict,
+                self._SERVICE_CLIENT_TIMEOUT_SECONDS,
+            )
+            return self.ingest_from_blob(blob_descriptor, ingestion_properties=ingestion_properties)
```

Comment:
You don't need this to be inside the "with", right? once the file is uploaded you can release it

Reply:
StreamDescriptor is in memory anyway, it should be closed at the very end (unless we need to re-use it, like in ManagedStreaming), but it matters much less.

FileDescriptor is an actual file on the disk, and nobody can use it (depends how you open it) until it's closed, so it's better to close them asap.

Reply:
Great, thanks - changed